### DISCUSSION TITLE
Should executeMutation be part of the public api docs, similar to fetchQuery?

### DISCUSSION THREAD

**Discussion by jgrisafe** [TanStack/query#4664] [Polls]
I was working with my team and noticed they were using useMutation inside services, nested in business logic and completely detached from the react lifecycle. They were plucking the 'mutator' object off of the return value, passing it to services, and using it to perform asynchronous mutations (which they said they learned from the docs). In my opinion, this is not a pattern we want devs to get used to. The "useSomething" notation is meant for react hooks, and is named as such to remind people that (normally) they cannot be run outside of a react component. They should be used when they need to be rerun on state changes, not for performing one-off mutations triggered by user interaction. I think it might be wise to push the use of executeMutation for these types of instances.

**Comment by TkDodo** [comment]
We had it in v3, but removed it in v4 because it wasn't documented and wasn't useful for most people. You can relatively easily implement the behaviour for yourself, as shown in the migration guide:

https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#removed-undocumented-methods-from-the-queryclient-query-and-mutation

**Reply by rendro** [reply]
This saved my day!
